impl/src/test/java/org/ehcache/impl/internal/events/InvocationScopedEventSinkTest.java

```diff
@@ -73,4 +80,46 @@ public void testReset() {
     verifyNoMoreInteractions(listener);
   }
 
+  /**
+   * Make sure an interrupted sink sets the interrupted flag and keep both event queues in the state
+   * as of before the event that was interrupted.
+   *
+   * @throws InterruptedException
+   */
+  @Test
+  public void testInterruption() throws InterruptedException {
```

Comment:
We claimed this was broken right?  We figured the `removeLast()` call was wrong in a multithreaded environment?

Reply:
I thought there should be not `removeLast()`. But in fact, it's ok. Since the event is already in `events` on interruption, it needs to be removed. I realized that while coding the test.

Then, is the class supposed to be thread-safe? Or thread local? I thought it was thread local.